src/gh_mirrors.rs

```diff
@@ -21,9 +22,11 @@ pub fn gh_url_to_org_and_name(url: &str) -> Result<(String, String)> {
     Ok((org.to_string(), name.to_string()))
 }
 
-pub fn fetch(url: &str) -> Result<()> {
-    let dir = repo_dir(url)?;
-    git::shallow_clone_or_pull(url, &dir)
+pub fn fetch(url: String) -> future::BoxFuture<(), Error> {
+    Box::new(future::lazy(move || {
+                              let dir = repo_dir(&url)?;
+                              git::shallow_clone_or_pull(&url, &dir)
+                          }))
```

Comment:
This use of futures seems slightly gratuitous and doesn't add anything. I think this would make more sense if the use of futures were pushed higher or lower in the stack, but just wrapping this function in `future::lazy` seems like code churn.